ui/component/invited/view.jsx

```diff
@@ -7,229 +7,178 @@ import ClaimPreview from 'component/claimPreview';
-              </div>
-
-              <div className="section__actions" style={{ marginTop: 'var(--spacing-m)' }}>
-                <Button
-                  button="primary"
-                  label={hasVerifiedEmail ? __('Finish Account') : __('Create Account')}
-                  navigate={`/$/${PAGES.AUTH}?redirect=/$/${PAGES.INVITE}/${referrer}`}
-                />
-                <Button button="link" label={__('Skip')} onClick={handleDone} />
-              </div>
-              <hr style={{ marginTop: 'var(--spacing-m)' }} />
+  const cardProps = {};
 
-              <div className="section">
-                <ChannelContent uri={fullUri} defaultPageSize={3} defaultInfiniteScroll={false} />
-              </div>
-            </div>
-          )
-        }
-        actions={
-          <div className="section__actions">
+  if (referrerSetError === ERRORS.ALREADY_CLAIMED) {
+    Object.assign(cardProps, {
+      title: __('Whoa!'),
+      subtitle: referrerIsChannel
+        ? __("You've already claimed your referrer, but we've followed this channel for you.")
+        : __('You already claimed your reward.'),
+      actions: <Button button="primary" label={__('Done!')} onClick={handleDone} />,
+    });
+  } else if (referrerSetError && hasUnclaimedRefereeReward) {
+    Object.assign(cardProps, {
+      title: __('Welcome!'),
+      subtitle: __(
+        'Something went wrong with your invite link. You can set and claim your invite reward after signing in.'
+      ),
+      actions: (
+        <>
+          <p className="error__text">{__('Not a valid invite')}</p>
+          <div className="card__actions">
             <Button
               button="primary"
-              label={hasVerifiedEmail ? __('Finish Account') : __('Create Account')}
-              navigate={`/$/${PAGES.AUTH}?redirect=/$/${PAGES.INVITE}/${referrer}`}
+              label={hasVerifiedEmail ? __('Verify') : __('Create Account')}
+              navigate={`/$/${PAGES.AUTH}?redirect=/$/${PAGES.REWARDS}`}
             />
-            <Button button="link" label={__('Skip')} onClick={handleDone} />
+            <Button button="link" label={__('Explore')} onClick={handleDone} />
           </div>
-        }
+        </>
+      ),
+    });
+  } else if (!rewardsApproved) {
+    const signUpButton = (
+      <Button
+        button="link"
+        label={hasVerifiedEmail ? __('Finish verification') : __('Create an account')}
+        navigate={`/$/${PAGES.AUTH}?redirect=/$/${PAGES.INVITE}/${referrerUri || ''}`}
       />
     );
+
+    Object.assign(cardProps, {
+      title: referrerIsChannel
+        ? __('%channel_name% invites you to the party!', { channel_name: channelTitle })
+        : __("You're invited!"),
+      subtitle: referrerIsChannel ? (
+        <I18nMessage tokens={{ channel_name: channelTitle, signup_link: signUpButton, SITE_NAME }}>
+          %channel_name% is waiting for you on %SITE_NAME%. Create your account now.
+        </I18nMessage>
```

Comment:
:small_red_triangle: Hm ... `channelTitle` is undefined for some channels (e.g. `@t2:2`), so the sentence could potentially be odd?

Fallback to claim name?

Reply:
`channelTitle` comes from `getChannelTitleFromClaim` which already falls back to name in undefined cases